platform/macos/src/MGLMapView.mm

```diff
@@ -1402,10 +1383,10 @@ - (void)handlePanGesture:(NSPanGestureRecognizer *)gestureRecognizer {
         _mbglMap->cancelTransitions();
 
         if (gestureRecognizer.state == NSGestureRecognizerStateBegan) {
-            _scaleAtBeginningOfGesture = _mbglMap->getScale();
+            _zoomAtBeginningOfGesture = _mbglMap->getZoom();
         } else if (gestureRecognizer.state == NSGestureRecognizerStateChanged) {
-            CGFloat newZoomLevel = log2f(_scaleAtBeginningOfGesture) - delta.y / 75;
-            [self scaleBy:powf(2, newZoomLevel) / _mbglMap->getScale() atPoint:startPoint animated:NO];
+            CGFloat newZoomLevel = _zoomAtBeginningOfGesture - delta.y / 75;
```

Comment:
The double-tap-drag gesture behaves similarly on iOS.

In case it helps, the iOS gestures are documented [in jazzy docs](https://www.mapbox.com/ios-sdk/api/3.5.2/gesture-recognizers.html), and the macOS gestures are listed in macosapp under Help > Mapbox GL Help.